sormas-ui/src/main/java/de/symeda/sormas/ui/security/config/DefaultOpenIdAuthenticationDefinition.java

```diff
@@ -0,0 +1,176 @@
+import fish.payara.security.annotations.ClaimsDefinition;
+import fish.payara.security.annotations.OpenIdAuthenticationDefinition;
+import fish.payara.security.annotations.OpenIdProviderMetadata;
+import fish.payara.security.openid.api.DisplayType;
+import fish.payara.security.openid.api.OpenIdConstant;
+import fish.payara.security.openid.api.PromptType;
+
+import java.lang.annotation.Annotation;
+
+/**
+ * Default implementation of  {@link OpenIdAuthenticationDefinition} to allow inline declaration.
+ * <p/>
+ * Configurations to be provided trough Payara MP.
+ *
+ * @author Alex Vidrean
+ * @since 12-Aug-20
+ */
+public class DefaultOpenIdAuthenticationDefinition implements OpenIdAuthenticationDefinition {
+
+	@Override
+	public String providerURI() {
+		return "";
+	}
+
+	@Override
+	public OpenIdProviderMetadata providerMetadata() {
+		return new OpenIdProviderMetadata() {
+
+			@Override
+			public Class<? extends Annotation> annotationType() {
+				return OpenIdProviderMetadata.class;
+			}
+
+			@Override
+			public String authorizationEndpoint() {
+				return "";
+			}
+
+			@Override
+			public String tokenEndpoint() {
+				return "";
+			}
+
+			@Override
+			public String userinfoEndpoint() {
+				return "";
+			}
+
+			@Override
+			public String jwksURI() {
+				return "";
+			}
+		};
+	}
+
+	@Override
+	public ClaimsDefinition claimsDefinition() {
+		return new ClaimsDefinition() {
+
+			@Override
+			public Class<? extends Annotation> annotationType() {
+				return ClaimsDefinition.class;
+			}
+
+			@Override
+			public String callerNameClaim() {
+				return OpenIdConstant.PREFERRED_USERNAME;
+			}
+
+			@Override
+			public String callerGroupsClaim() {
+				return OpenIdConstant.GROUPS;
+			}
+		};
+	}
+
+	@Override
+	public String clientId() {
+		return "";
+	}
+
+	@Override
+	public String clientSecret() {
+		return "";
+	}
+
+	@Override
+	public String redirectURI() {
+		return "${baseURL}/Callback";
+	}
+
+	@Override
+	public String[] scope() {
+		return new String[] { OpenIdConstant.OPENID_SCOPE };
+	}
+
+	@Override
+	public String responseType() {
+		return "code";
+	}
+
+	@Override
+	public String responseMode() {
+		return "";
+	}
+
+	@Override
+	public PromptType[] prompt() {
+		return new PromptType[] {};
+	}
+
+	@Override
+	public DisplayType display() {
+		return DisplayType.PAGE;
+	}
+
+	@Override
+	public boolean useNonce() {
+		return true;
+	}
+
+	@Override
+	public boolean useSession() {
+		return true;
+	}
+
+	@Override
+	public String[] extraParameters() {
+		return new String[] {};
+	}
+
+	@Override
+	public Class<? extends Annotation> annotationType() {
+		return fish.payara.security.annotations.OpenIdAuthenticationDefinition.class;
+	}
+
+	@Override
+	public int jwksConnectTimeout() {
+		return 500;
+	}
+
+	@Override
+	public int jwksReadTimeout() {
+		return 500;
+	}
+
+	@Override
+	public boolean tokenAutoRefresh() {
+		return false;
+	}
+
+	@Override
+	public int tokenMinValidity() {
+		return 10 * 1000;
```

Comment:
shouldn't these values be extracted somewhere in a properties file?

Reply:
The values in this class are just the default values copied from `OpenIdAuthenticationDefinition` and they can be overwritten trough MicroProfile Config Properties.

In this case the `tokenMinValidity` can be overwritten by setting the `payara.security.openid.token.minValidity` property.